concrete/src/User/UserServiceProvider.php

```diff
@@ -26,5 +31,37 @@ public function register()
         $this->app->bind('Concrete\Core\User\Avatar\AvatarServiceInterface', function() use ($app) {
             return $app->make('user/avatar');
         });
+
+        // Handle binding events
+        if ($this->app->resolved(EventDispatcher::class)) {
+            $this->bindEvents($this->app->make(EventDispatcher::class));
+        } else {
+            $this->app->extend(EventDispatcher::class, function(EventDispatcher $director) {
+                $this->bindEvents($director);
+                return $director;
+            });
+        }
+    }
+
+    protected function bindEvents(EventDispatcherInterface $dispatcher)
+    {
+        $dispatcher->addListener('on_after_user_deactivate', function($e) {
+            $this->app->call([$this, 'handleEvent'], ['event' => $e]);
+        });
+    }
+
+    /**
+     * Handle routing bound events
+     *
+     * @param \Symfony\Component\EventDispatcher\Event $event
+     * @param \Concrete\Core\User\Notification\UserNotificationEventHandler $service
+     */
+    public function handleEvent(Event $event, UserNotificationEventHandler $service)
+    {
+        $entity = $event->getUserEntity();
+
+        if ($event instanceof DeactivateUser && $entity) {
```

Comment:
Can we have situations when `$event` is not an instance of `DeactivateUser`?
- if no: this check is useless
- if yes: `$event` may not have a `getUserEntity` method (so I'd move the `instanceof` check at the beginning of the method).

Reply:
Yes, this method is public. I agree it needs to move up :+1: